lighthouse-core/test/gather/driver-test.js

```diff
@@ -166,8 +180,9 @@ describe('.getObjectProperty', () => {
 
 describe('.getRequestContent', () => {
   it('throws if getRequestContent takes too long', async () => {
+    // @ts-ignore
     connectionStub.sendCommand = jest.fn()
-      .mockImplementationOnce(() => new Promise(r => setTimeout(r), 5000));
+      .mockImplementationOnce(() => new Promise(r => setTimeout(r, 5000)));
```

Comment:
this is why types are important

Reply:
I'm now very concerned how this was working before, wouldn't this have resolved immediately and we shouldn't have gotten protocol_timeout? 🤔

Reply:
https://github.com/GoogleChrome/lighthouse/pull/10135
tldr: microtasks